src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -747,17 +748,19 @@ export class PostgresQueryRunner extends BaseQueryRunner implements QueryRunner
                     downQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" DROP DEFAULT`));
                 }
 
-                // remove old ENUM
-                upQueries.push(this.dropEnumTypeSql(table, newColumn, oldEnumName));
-                downQueries.push(this.createEnumTypeSql(table, oldColumn, oldEnumName));
+                // up: remove old renamed enum type
+                // down: rename new enum type and create old enum type
+                upQueries.push(this.dropEnumTypeSql(table, newColumn, prevEnumNameWithOldSuffixAndSchema));
+                downQueries.push(this.createEnumTypeSql(table, oldColumn));
+                downQueries.push(new Query(`ALTER TYPE ${newEnumNameWithSchema} RENAME TO ${newEnumNameWithOldSuffix}`));
             }
 
             if (oldColumn.isNullable !== newColumn.isNullable) {
                 if (newColumn.isNullable) {
-                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${oldColumn.name}" DROP NOT NULL`));
+                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" DROP NOT NULL`));
                     downQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${oldColumn.name}" SET NOT NULL`));
                 } else {
-                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${oldColumn.name}" SET NOT NULL`));
+                    upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" SET NOT NULL`));
```

Comment:
Old column to new column via diff.  Intentional?

Reply:
yeah, see above - i think lol. need to review my old code or maybe write some extra tests to make sure haha.

Reply:
Ok - once you're ready, re-request my review.